### PR TITLE
add dprime

### PR DESCRIPTION
**Pull request by leoscholl**
maybe useful signal-to-noise ratio function

### CODE REVIEW

File: aopy/analysis/tuning.py [hunk 1]
```diff
@@ -146,3 +146,34 @@ def run_tuningcurve_fit(mean_fr, targets, fit_with_nans=False, min_data_pts=3):
...
+
+    Returns:
+        (nch): d-prime value for each channel or unit.
+    
+    Examples:
+    
+        $d'$ is essentially a signal-to-noise ratio. You can think of the numerator as the maximum distance 
+        between two signals, while the denominator is the average noise at each signal. For example, in a two-target 
+        classification, the numerator becomes exactly the difference in signal between the two targets. Values of 
+        1, 2, and 3 correspond to correctly classifying the two targets at 69, 84, and 93 percent correct, respectively.
```
Comment (rcanfield2) [inline aopy/analysis/tuning.py:173]: The 69, 84, 93 are from the STD values? Its probably worth mentioning that more explicitly.

File: tests/test_analysis.py [hunk 2]
```diff
@@ -266,6 +267,42 @@ def test_get_mean_fr_per_condition(self):
...
+        signal_dist = np.array([[1, 2], [1, 2]]).T
+        print(noise_dist.shape)
+        dprime = aopy.analysis.calc_dprime(noise_dist, signal_dist)
+
+        np.testing.assert_allclose(dprime, np.array([1, 1]))
+
+        # Multi-class test
+        noise_dist = np.array([[0, 1, 0, 1], [0, 1, 0, 1]]).T
+        signal_dist = np.array([[1, 2, 1, 2], [2, 3, 2, 3]]).T
+        dprime = aopy.analysis.calc_dprime(noise_dist, signal_dist)
```
Comment (rcanfield2) [inline tests/test_analysis.py:302]: Its unclear to me what this is testing that is different from the one above (simple multi-channel test)

File: aopy/analysis/tuning.py [hunk 3]
```diff
@@ -146,3 +146,34 @@ def run_tuningcurve_fit(mean_fr, targets, fit_with_nans=False, min_data_pts=3):
...
+    
+    .. math::
+
+        d' = \\frac{µ_{max} - µ_{min}}{\sqrt{\sum_{i=0}^{n-1} (p_i)\sigma_i^2}}
+
+    where $µ_{max}$ and $µ_{min}$ are the mean of the two categories, $p_i$ is the proportion of trials
+    in the i-th category, and $\sigma_i^2$ is the standard deviation of the i-th category.
+
+    Args:
+        *dist (ntr, nch): distribution of the data for each category. d-prime is calculated along the first axis.
```
[the remark this answers is not in the document]
Comment (leoscholl) [reply aopy/analysis/tuning.py:163]: added examples and updated the description

File: aopy/analysis/tuning.py (hunk 1 above)
Comment (leoscholl) [reply aopy/analysis/tuning.py]: i updated

File: tests/test_analysis.py (hunk 2 above)
Comment (leoscholl) [reply tests/test_analysis.py]: yeah my mistake, i meant to add another distribution but i just added more trials within each distribution

**Review by rcanfield2 — APPROVED** [review]

**Pull request merged**